conftest.py

```diff
@@ -1,14 +1,330 @@
+                    return random.sample(parameter_dicts, num_samples)
+
+            elif isinstance(sampling, float):
+                if sampling < 0 or sampling > 1:
+                    raise ValueError(
+                        f"Portion of parameter dicts to sample must be "
+                        f"in the range [0, 1], got {sampling}."
+                    )
 
-    # Ensure graphblas-opt is built
-    subprocess.run(["python", os.path.join("mlir_graphblas", "src", "build.py")])
+                def sampling_method(parameter_dicts):
+                    parameter_dicts = list(parameter_dicts)
+                    num_samples = int(len(parameter_dicts) * sampling)
+                    return random.sample(parameter_dicts, num_samples)
 
+            elif sampling == "exhaustive":
+
+                def sampling_method(parameter_dicts):
+                    return parameter_dicts
+
+            else:
+                raise ValueError(
+                    f"{repr(sampling)} is not a supported sampling method."
+                )
+
+            # Grab all possible parameter dicts
+            parameter_names = parameter_choices.keys()
+            parameter_value_tuples = itertools.product(*parameter_choices.values())
+            all_parameter_dicts = (
+                dict(zip(parameter_names, parameter_values))
+                for parameter_values in parameter_value_tuples
+            )
+
+            # Find the requested parameter dicts
+            parameter_dicts = filter(parameter_dict_filter, all_parameter_dicts)
+            parameter_dicts = sampling_method(parameter_dicts)
+
+            # Append one parameter dict for each test case
+            for parameter_dict in parameter_dicts:
+                parameter_tuples.append(
+                    (
+                        generate_test_id_string(template_file, parameter_dict),
+                        (
+                            mlir_template.render(**parameter_dict),
+                            test_execution_command_template,
+                            template_file,
+                            parameter_dict,
+                        ),
+                    )
+                )
+
+    return parameter_tuples
+
+
+def generate_test_id_string(template_file: str, parameter_dict: Dict[str, str]) -> str:
+    return "".join(c for c in template_file if c.isalnum()) + "".join(
+        f"({re.escape(key)}:{re.escape(parameter_dict[key])})"
+        for key in sorted(parameter_dict.keys())
+    )
+
+
+def parameterize_templatized_filecheck_test(metafunc: "_pytest.python.Metafunc"):
+    sampling_method_string = metafunc.config.getoption("--filecheck-sampling")
+    if sampling_method_string.isdigit():
+        sampling = int(sampling_method_string)
+    elif re.match("^\d+(\.\d+)?$", sampling_method_string):
+        sampling = float(sampling_method_string)
+    else:
+        sampling = sampling_method_string
+    seed = (
```

Comment:
I should probably add a code comment here saying what this is doing. 

All the workers need to have run the exact same tests. Thus, they need to use the exact same random seed when sampling the tests (all the workers redundantly do this; not sure how to avoid this redundant work). I do this by making the `testrunuid` the seed (via all the `zlib.adler32` stuff) if the worker is running this code. Since the main/non-worker process doesn't run any tests (so nothing bad happens if we randomly sample different tests) and since the main process's `metafunc.config` doesn't have a `workerinput` attribute, I just use `time.time()`.

Reply:
Agree that it's a bit weird, but as long as it's well documented for *why* it behaves this way and what it attempting to accomplish, I'm fine with it. Probably just copy-paste exactly what you wrote here and put it in a comment in the code.

Reply:
I added the comment.